Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,204 @@
+fit_and_score <- structure(function(target.mat, feature.mat, 
+                                    parameters, n_folds = 3, scorer = NULL, 
+                                    pruning = TRUE){
+  ### total length
+  l <- nrow(target.mat)
+  
+  ### shuffle the data 
+  rand <- sample(nrow(target.mat))
+  target.mat <- target.mat[rand,]
+  feature.mat <- feature.mat[rand,]
+  
+  
+  ### Create n equally size folds
+  folds <- cut(seq(1,nrow(target.mat)), breaks = n_folds, labels = FALSE)
+  
+  ### train and test index for each fold
+  fold_split_idx <- NULL
+  
+  ### segment the data into test and train
+  for(i in 1:n_folds){
+    #Segement your data by fold using the which() function 
+    fold_split_idx$test <- rbind(fold_split_idx$test, which(folds == i, arr.ind = TRUE))
+    fold_split_idx$train <- rbind(fold_split_idx$train, which(folds != i, arr.ind = TRUE))
+    
+  }
+  
+  ### fold trees
+  fold_tree <- NULL
+  for(i in 1:n_folds){
+    fold_tree[[i]] <- mmit(target.mat[fold_split_idx$train[i,],], feature.mat[fold_split_idx$train[i,],],  
+                                       max_depth = as.numeric(parameters$max_depth), margin = as.numeric(parameters$margin), 
+                                       loss = parameters$loss, min_sample = as.numeric(parameters$min_sample))
+  }
+  
+  ### master tree
+  master_tree <- mmit(target.mat, feature.mat,  
+                      max_depth = as.numeric(parameters$max_depth), margin = as.numeric(parameters$margin), 
+                      loss = parameters$loss, min_sample = as.numeric(parameters$min_sample))
+  
+ 
+  ### for traning data
+  alpha_train_scores <- NULL
+  alpha_train_objective_values <- NULL
+  
+  ### if pruning
+  if(pruning){
+    # Get the pruned master and cross-validation trees
+    master_data <- mmit.pruning(master_tree)
+    master_alphas <- rev(unlist(lapply(master_data, function(x) x$alpha)))
```

Comment:
I don't understand why you reverse the alphas and the trees. Can you explain please?

Reply:
I fact, I think that reverse sorting the alphas does not make sense and breaks the pruning code.

Reply:
See comment at line 83.